src/matrix-utils.ts

```diff
@@ -76,9 +76,13 @@ function waitForSync(client: MatrixClient) {
 function secureRandomString(entropyBytes: number): string {
   const key = new Uint8Array(entropyBytes);
   crypto.getRandomValues(key);
+  // encode to base64url as this value goes into URLs
   return btoa(
     key.reduce((acc, current) => acc + String.fromCharCode(current), "")
-  ).replace(/=*$/, "");
+  )
+    .replace("+", "-")
+    .replace("/", "_")
+    .replace(/=*$/, "");
```

Comment:
Is there a way to have a more generic function that guarantees that all url incompatible params get replaced?
Or do we know with certainty that those are all the possible once?

Reply:
(synced with Dave: seems like there isnt something really great.)